src/main/scala/chisel3/util/util.scala

```diff
@@ -11,4 +11,15 @@ package object util {
   type ValidIO[+T <: Data] = chisel3.util.Valid[T]
   val ValidIO = chisel3.util.Valid
   val DecoupledIO = chisel3.util.Decoupled
+
+  implicit class DecoupledExtensions[A <: Data](x: DecoupledIO[A]) {
+    def map[B <: Data](f: A => B): DecoupledIO[B] = {
+      val res = f(x.bits)
```

Comment:
What if this were called "bits" and wire was called "res"? Just looking at the names of the resulting signals

Reply:
Or "res_bits" and "res"

Reply:
Alternatively, it could just be `wire.bits := f(x.bits)`, or I think I like `map` better as a name because it indicates that this came from calling `.map`.

Also, this might be a good opportunity to utilize https://github.com/chipsalliance/chisel3/pull/2580 by using `_` as the leading character for names in this block.

Reply:
Use of the leading underscore seems good to me. Maybe `_map` or `_mapped`. Alternatively, don't create a `val res` here and just inline it. That'll create an unnamed node and that should get inlined? That would then motivate changing `val wire` to `val _map`.

Can you generate some examples of what the output looks like in Verilog for this?